__tests__/server.test.js

```diff
@@ -0,0 +1,56 @@
+const request = require("supertest");
+const server = require("../middleware/server");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data/index.js");
+
+afterAll(() => {
+  return db.end();
+});
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+describe("server --> nc_news_test", () =>{
+    describe("General error handling", () =>{ 
+        test("Returns status code 404 to request to non-existant endpoint", () => {
+            return request(server).get("/api/thisDoesNotExist").expect(404);
+        });
+        test("Returns error message to request to non-existant endpoint", () => {
+            return request(server)
+              .get("/api/thisDoesNotExist")
+              .then((result) => {
+                expect(result.error.text).toEqual("Invalid URL");
+              });
+        });
```

Comment:
Good work implementing this custom 404 handler for any path which does not exist.

As this is an integration test, I would recommend just having 1 test which checks everything - the status code 404, and the response.